client/allocrunner/networking_cni.go

```diff
@@ -184,12 +184,15 @@ func (c *cniNetworkConfigurator) cniToAllocNet(res *cni.Result) (*structs.AllocN
 
 	}
 
-	// Use the first DNS results.
+	// Use the first DNS results, if non-empty
```

Comment:
Maybe not the best place for this, but it may be useful to have a place listing the multitude of possibilities for DNS config sources 😅